providers/package.rb

```diff
@@ -12,24 +12,47 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-Chef::Resource::Execute.send(:include, Cygwin::Helpers)
+include Cygwin::Helpers
+
+def whyrun_supported?
+  true
+end
+
+def load_current_resource
+  @current_resource = Chef::Resource::CygwinPackage.new(@new_resource.name)
```

Comment:
I would like to see this changed to use the new syntax, newing up the resource based on its generated name is deprecated.

Something like

`Chef::Resource.resource_for_node(:cygwin_package, node).new(new_resource.name, run_context)`